### PR TITLE
:tada: New Source: CallRail [low-code-cdk]

### PR DESCRIPTION
**Pull request by asuliman17**
## What
The following PR adds the new low code CallRail Connector. 

## How
This PR uses the lowcode CDK to create a CallRail connector with the following streams:

calls
companies
text_messages
trackers
users

## Recommended reading order
1. `spec.yaml`
2. `calrail.yaml`
3. `schemas`

## 🚨 User Impact 🚨
No breaking changes.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Hello @adhamsuliman, Marcos from Airbyte here 👋 . We received more than 25 new contributions along the weekend. One is yours 🎉  thank so much for! Our team is limited and maybe the review process can take longer than expected. As described in the Airbyte's Hacktoberfest your contribution was submitted before November 2nd and it is eligible to win the prize. The review process will validate other requirements. I ask to you patience until someone from the team review it. 

Because I reviewed some contributions for Hacktoberfest so far I saw some common patterns you can check in advance:
* [ ] Make sure you have added connector documentation to `/docs/integrations/`
* [ ] Remove the file `catalog` from `/integration_tests`
* [ ] Edit the `sample_config.json` inside `/integration_tests`
* [ ] For the `configured_catalog` you can use only `json_schema: {}`
* [ ] Add title to all properties in the `spec.yaml`
* [ ] Make sure the `documentationUrl` in the `spec.yaml` redirect to Airbyte's future connector page, eg: connector Airtable the `documentationUrl: https://docs.airbyte.com/integrations/sources/airtable`
* [ ] Review now new line at EOF (end-of-file) for all files.

If possible send to me a DM in Slack with the tests credentials, this process will make easier to us run integration tests and publish your connector. If you only has production keys, make sure to create a bootstrap.md explaining how to get the keys.

**Comment by marcosmarxm** [comment]
@adhamsuliman can you share the output of integration tests?

**Comment by adhamsuliman** [comment]
> 

@marcosmarxm 

I'm passing all tests except for backward compatibility. I haven't changed anything with the schema in between docker builds, so I can't imagine as to why I would have this error. Could you please advise how to resolve them? 


image

**Comment by marcosmarxm** [comment]
Hello! I'm going to be out of the office this Friday and won't be able to review your contribution again today, I return next Monday. So far, most contributions look solid and are almost done to be approved. As said in [Chris' comment](https://airbytehq.slack.com/archives/C0450PCSDB2/p1667430678866629) all contributions made before 2-November are eligible to receive the prize and have 2 weeks to merge the contributions. But I ensure next week we're going to have your contribution merged. If you have questions about the implementation you can send them in `#hacktoberfest-2022` in Airbyte's Slack.

Sorry the inconvenience and see you again next week, thank you so much for your contribution!

File: airbyte-integrations/connectors/source-callrail/source_callrail/callrail.yaml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    $options:
+      url_base: "https://api.callrail.com/v3/a/"
+    name: "{{ options['name'] }}"
+    primary_key: "{{ options['primary_key'] }}"
+    record_selector:
+      extractor:
+        type: DpathExtractor
+        field_pointer: [ "{{ options['name'] }}" ]
+    paginator:
+      type: NoPagination
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-callrail/source_callrail/callrail.yaml:46]: Please implement the correct pagination for this connector.

File: airbyte-integrations/connectors/source-callrail/source_callrail/callrail.yaml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    schema_loader:
+      $ref: "*ref(definitions.schema_loader)"
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+      requester:
+        $ref: "*ref(definitions.requester)"
+        path:  "{{ config['account_id'] }}/calls.json?"
+        request_options_provider:
+          request_parameters:
+            fields: "call_type,company_id,company_name,company_time_zone,created_at,device_type,first_call,formatted_call_type,formatted_customer_location,formatted_business_phone_number,formatted_customer_name,prior_calls,formatted_customer_name_or_phone_number,formatted_customer_phone_number,formatted_duration,formatted_tracking_phone_number,formatted_tracking_source,formatted_value,good_lead_call_id,good_lead_call_time,lead_status,note,source,source_name,tags,total_calls,value,waveforms,tracker_id,speaker_percent,keywords,medium,campaign,referring_url,landing_page_url,last_requested_url,referrer_domain,utm_source,utm_medium,utm_term,utm_content,utm_campaign,utma,utmb,utmc,utmv,utmz,ga,gclid,fbclid,msclkid,milestones,timeline_url,keywords_spotted,call_highlights,agent_email,keypad_entries"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-callrail/source_callrail/callrail.yaml:64]: This is required? Please remove the fields and ingest all fields supported by the API.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments. The error you posted in about the test not finding the file `conversations.json`

File: airbyte-integrations/connectors/source-callrail/source_callrail/callrail.yaml (hunk 1 above)
Comment (adhamsuliman) [reply airbyte-integrations/connectors/source-callrail/source_callrail/callrail.yaml]: I've added pagination in my latest commit with hash:  e105d10cb2f8af9fbc322139b1d6192152a7f219.

File: airbyte-integrations/connectors/source-callrail/source_callrail/callrail.yaml (hunk 2 above)
Comment (adhamsuliman) [reply airbyte-integrations/connectors/source-callrail/source_callrail/callrail.yaml:74]: These fields are "Additional User Requested Response Fields" that will only be returned if explicitly called out. You can find these fields within the [API documentation](https://apidocs.callrail.com/#listing-all-calls).

**Comment by adhamsuliman** [comment]
> Some comments. The error you posted in about the test not finding the file `conversations.json`

Hmmm. I can't seem to determine why it can't find the file. I can see the `conversations.json` under the schemas directory. Any advice on how to debug this issue

**Review by marcosmarxm — APPROVED** [review]
@adhamsuliman please make sure you have allow me to push changes to your branch.

**Comment by marcosmarxm** [comment]
I'll finish the release process tomorrow.

**Comment by marcosmarxm** [comment]
Merged at #19407

**Pull request closed without merging**